packages/extension-base/src/services/keyring-service/context/handlers/Migration.ts

```diff
@@ -0,0 +1,249 @@
+      }
+
+      rs[upcomingProxyId].push(item);
+
+      return rs;
+    }, {});
+  }
+
+  public accountProxiesToEligibleSoloAccountMap (accountProxyMap: Record<string, AccountProxy[]>): Record<string, SoloAccountToBeMigrated[]> {
+    const eligibleSoloAccountMap: Record<string, SoloAccountToBeMigrated[]> = {};
+
+    Object.entries(accountProxyMap).forEach(([upcomingProxyId, accounts]) => {
+      eligibleSoloAccountMap[upcomingProxyId] = accounts.map((account) => {
+        return {
+          upcomingProxyId,
+          proxyId: account.accounts[0].proxyId,
+          address: account.accounts[0].address,
+          name: account.name,
+          chainType: account.chainTypes[0]
+        } as SoloAccountToBeMigrated;
+      });
+    });
+
+    return eligibleSoloAccountMap;
+  }
+
+  public sortUAByParent (accountProxies: AccountProxy[]): AccountProxy[] {
+    const undefinedToStr = (str: string | undefined) => str ?? '';
+
+    return accountProxies.sort((a, b) => undefinedToStr(a.parentId) < undefinedToStr(b.parentId) ? -1 : undefinedToStr(a.parentId) > undefinedToStr(b.parentId) ? 1 : 0);
+  }
+
+  public migrateSoloToUnifiedAccount (request: RequestMigrateSoloAccount): ResponseMigrateSoloAccount {
+    const { accountName, sessionId, soloAccounts } = request;
+    const password = this.sessionIdToPassword[sessionId].password;
+
+    keyring.unlockKeyring(password);
+    this.parentService.updateKeyringState();
+
+    const modifiedPairs = structuredClone(this.state.modifyPairs);
+
+    const firstAccountInfo = soloAccounts[0];
+    const upcomingProxyId = firstAccountInfo.upcomingProxyId;
+    const firstAccountOldProxyId = firstAccountInfo.proxyId;
+    const mnemonic = this.parentService.context.exportAccountProxyMnemonic({ password, proxyId: firstAccountOldProxyId }).result;
+
+    const keypairTypes = SUPPORTED_ACCOUNT_CHAIN_TYPES.map((chainType) => getDefaultKeypairTypeFromAccountChainType(chainType as AccountChainType));
+
+    keypairTypes.forEach((type) => {
+      const suri = getSuri(mnemonic, type);
+      const pair = keyring.createFromUri(suri, {}, type);
+      const address = pair.address;
+
+      modifiedPairs[address] = { accountProxyId: upcomingProxyId, migrated: true, key: address };
+    });
+
+    this.state.upsertAccountProxyByKey({ id: upcomingProxyId, name: accountName });
+
+    this.migrateDerivedSoloAccountRelationship(soloAccounts);
+
+    this.state.upsertModifyPairs(modifiedPairs);
+    keypairTypes.forEach((type) => {
+      const suri = getSuri(mnemonic, type);
+      const { derivePath } = keyExtractSuri(suri);
+      const metadata = {
+        name: accountName,
+        derivationPath: derivePath ? derivePath.substring(1) : undefined
+      };
+
+      const rs = keyring.addUri(suri, metadata, type);
+      const address = rs.pair.address;
+
+      this.state._addAddressToAuthList(address, true);
+    });
+
+    keyring.lockAll(false);
+    this.parentService.updateKeyringState();
```

Comment:
do we need to delete the old pairs?

Reply:
the new pairs are upserted (replace old pairs), so we do not need to delete